pkgs/development/compilers/haxe/default.nix

```diff
@@ -1,9 +1,23 @@
-{ lib, stdenv, fetchFromGitHub, coreutils, ocaml-ng, zlib, pcre, neko, mbedtls }:
+{ lib, stdenv, fetchFromGitHub, fetchurl, coreutils, ocaml-ng, zlib, pcre, neko, mbedtls }:
 
 let
+  defaultOCamlPackages = ocaml-ng.ocamlPackages.overrideScope'
+    (self: super: {
+      # haxe 4.0 and 4.1 unfortunately need an older version of extlib
+      ocaml_extlib-1-7-7 = super.ocaml_extlib.overrideAttrs (_: rec {
+        version = "1.7.7";
+        src = fetchurl {
+          url = "https://github.com/ygrek/ocaml-extlib/releases/download/${version}/extlib-${version}.tar.gz";
+          sha256 = "1sxmzc1mx3kg62j8kbk0dxkx8mkf1rn70h542cjzrziflznap0s1";
+        };
```

Comment:
You could reuse `fetchFromGitHub` here.

Reply:
No, that is a release asset and not the automatically generated archive, so we need to use `fetchurl`.

Reply:
Oh I see, nevermind then.